core/types/receipt.go

```diff
@@ -484,6 +485,9 @@ func (r *Receipt) DeriveFieldsV3ForSingleReceipt(txnIdx int, blockHash libcommon
 
 // TODO: maybe make it more prettier (only for debug purposes)
 func (r *Receipt) String() string {
-	str := fmt.Sprintf("Receipt of tx %+v", *r)
-	return str
+	j, err := json.Marshal(r)
+	if err != nil {
+		return fmt.Sprintf("Error during JSON marshalling, receipt: %v", *r)
```

Comment:
missing `+` in `%+v` (previously)
also do you want to wrap the json err with `%w`?

Reply:
Unable to use %w here since it is only allowed in fmt.Errorf, but I have used %s